UnityWeld/Binding/Internal/AdapterResolver.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Linq;
+using System.Reflection;
+using UnityWeld.Ioc;
+
+namespace UnityWeld.Binding.Internal
+{
+    /// <summary>
+    /// Helper class for creating adapters
+    /// </summary>
+    public static class AdapterResolver
+    {
+        private static IWeldContainerIoC container;
+
+        public static IAdapter CreateAdapter(Type adapterType)
+        {
+            if(container == null)
+            {
+                SetupWeldContainer();
+            }
+            return container.Resolve<IAdapter>(adapterType);
+        }
+
+        private static void SetupWeldContainer()
+        {
+            var containerTypes = TypeResolver.TypesWithWeldContainerAttribute.Where(t => typeof(IWeldContainerIoC).IsAssignableFrom(t));
+            var containerTypesCount = containerTypes.Count();
+            if (containerTypesCount > 0)
+            {
+                if(containerTypesCount > 1)
+                {
+                    UnityEngine.Debug.LogWarningFormat("You have multiple classes marked with the WeldContainer Attribute, only one can be used!");
```

Comment:
I think I'd actually prefer if this threw an exception, since it's undefined behaviour and failing fast would force users to fix the issue.